src/hybrid/parse/hybrid_callbacks.c

```diff
@@ -404,6 +407,42 @@ void handleFilter(ArgParser *parser, const void *value, void *user_data) {
     AGPLN_AddStep(ctx->plan, &stp->base);
 }
 
+// POLICY callback for FILTER clause - handles value mapping and attribute creation
+void handleFilterPolicy(ArgParser *parser, const void *value, void *user_data) {
+    ParsedVectorData *pvd = (ParsedVectorData*)user_data;
+    const char *policy = *(const char**)value;
+
+    const char *attrValue = policy;
+    // Map ADHOC to adhoc_bf (BATCHES is used as-is)
+    // Note: ADHOC_BF is already rejected by ARG_OPT_ALLOWED_VALUES
+    if (strcasecmp(policy, "ADHOC") == 0) {
+        attrValue = VECSIM_POLICY_ADHOC_BF;
```

Comment:
being picky, but if u have the if else on policy and u create the QueryAttribute there, u can skip an string duplication.